### PR TITLE
[#3024] improvement(IT): close containers and upload container log after all tests are finished

### PR DESCRIPTION
**Pull request by xiaozcy**
### What changes were proposed in this pull request?

close containers and upload container log after all tests are finished

### Why are the changes needed?

Fix: #3024 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

existing ITs

### CODE REVIEW

**Comment by xiaozcy** [comment]
@yuqi1129 This PR is ready for review. BTW, can you add `upload log` label to this PR so that we could verify if the logs are uploaded as expected?

**Comment by yuqi1129** [comment]
@xiaozcy 
It seems fine if all pipelines succeed, please update the code to test if everything is okay with a failure.

**Comment by xiaozcy** [comment]
> @xiaozcy It seems fine if all pipelines succeed, please update the code to test if everything is okay with a failure.

I use #3213 to test if logs are uploaded as expected when some tests failed, the results indicate that everything is fine in this situation.
#3213 should be closed after you check the results.

**Comment by yuqi1129** [comment]
> > @xiaozcy It seems fine if all pipelines succeed, please update the code to test if everything is okay with a failure.
> 
> @yuqi1129 I use #3213 to test if logs are uploaded as expected when some tests failed, the results indicate that everything is fine in this situation. #3213 should be closed after you check the results.

I see, thanks for your work.

File: integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/CloseContainerExtension.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+package com.datastrato.gravitino.integration.test.util;
+
+import com.datastrato.gravitino.integration.test.container.ContainerSuite;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CloseContainerExtension implements BeforeAllCallback {
```
Comment (yuqi1129) [inline integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/CloseContainerExtension.java:13]: @xiaozcy 
Adding a comment about this class is preferred.

**Comment by yuqi1129** [comment]
LGTM except for one small problem.

File: integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/CloseContainerExtension.java (hunk 1 above)
Comment (xiaozcy) [reply integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/CloseContainerExtension.java:18]: Done

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**